src/v/cluster/rm_stm.cc

```diff
@@ -1100,10 +1100,13 @@ rm_stm::replicate_tx(model::batch_identity bid, model::record_batch_reader br) {
         _mem_state.estimated.emplace(bid.pid, _insync_offset);
     }
 
+    auto ack = raft::consistency_level::leader_ack;
```

Comment:
Do not understand about  _log_state.inflight...
For what do we need it? Also when do you wanna update this map?

Reply:
existing code depends on `_mem_state`, `_mem_state` is reset on every term update (when leader election happens) with flushing on disk we want to avoid disruptions during the re-election so all accesses to `_mem_state` should be beyond the `!txn_ga` check